apps/web/pages/api/trpc/[trpc].ts

```diff
@@ -30,34 +30,60 @@ export default trpcNext.createNextApiHandler({
    * @link https://trpc.io/docs/caching#api-response-caching
    */
   responseMeta({ ctx, paths, type, errors }) {
-    // assuming we have all our public routes in `viewer.public`
-    const allPublic = paths && paths.every((path) => path.startsWith("viewer.public."));
-    // checking that no procedures errored
-    const allOk = errors.length === 0;
-    // checking we're doing a query request
-    const isQuery = type === "query";
-
-    // i18n response depends on request header
-    const nonCacheableQueries = ["viewer.public.i18n"];
-    const isThereANonCacheableQuery = paths?.some((path) => nonCacheableQueries.includes(path));
-    const isThereACacheableQuery = paths?.some((path) => !nonCacheableQueries.includes(path));
-    if (isThereANonCacheableQuery && isThereACacheableQuery) {
-      console.warn(
-        "Cacheable and Non-cacheable queries are mixed in the same request. Not going to cache the request"
-      );
-    }
+    // Some helpers relevant to this function only
+    const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+    const TWO_HOURS_IN_SECONDS = 60 * 60 * 2;
+
+    // Our response to indicate no caching
+    const noCacheResponse = {};
 
+    // TODO: Confirm this is doing what the original code intended
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore ctx.req is available for SSR but not SSG
-    if (!!ctx?.req && allPublic && allOk && isQuery && !isThereANonCacheableQuery) {
-      // cache request for 1 day + revalidate once every 5 seconds
-      const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
-      return {
-        headers: {
-          "cache-control": `s-maxage=5, stale-while-revalidate=${ONE_DAY_IN_SECONDS}`,
-        },
-      };
+    const isSSR = !!ctx?.req;
+    if (isSSR) {
+      return noCacheResponse;
+    }
+
+    // No caching if we have a non-public path
+    // Assuming we have all our public routes in `viewer.public`
+    if (!paths || !paths.every((path) => path.startsWith("viewer.public."))) {
+      return noCacheResponse;
     }
-    return {};
+
+    // No caching if we have any procedures errored
+    if (errors.length !== 0) {
+      return noCacheResponse;
+    }
+
+    // Never cache non-queries (aka mutations)
+    if (type !== "query") {
+      return noCacheResponse;
+    }
+
+    // TODO: Discuss whether this value achieves what the original comment below describes
+    // cache request for 1 day + revalidate once every 5 seconds
+    let cacheValue = `s-maxage=5, stale-while-revalidate=${ONE_DAY_IN_SECONDS}`;
```

Comment:
Maybe I'm not familiar with the specs but isn't this "cache request for 5 seconds + revalidate once every 1 day" ?

Reply:
I agree with you. I called this out in [this comment](https://github.com/calcom/cal.com/pull/6823#discussion_r1092556306). I decided to preserve the original comment and code as it was before my change.

Reply:
Let's fix it instead

Reply:
Will do - my comment is asking for clarity on what "fixed" means. Should the code be fixed to match the comment, or should the comment be fixed to match the code?

Reply:
Let's match the comment instead. Caching for 5 seconds doesn't make much sense IMO